packages/yaru_colors/lib/src/yaru_color_extension.dart

```diff
@@ -0,0 +1,45 @@
+import 'package:flutter/widgets.dart';
+
+/// Set of useful methods when working with [Color]
+extension YaruColorExtension on Color {
+  /// Makes the color lighter by the given [amount], relatively to the current color luminosity.
+  /// [amount] value must be clamped between 0.0 and 1.0
+  Color lighten(double amount) {
+    assert(amount >= 0.0 && amount <= 1.0);
+    return _lightness(amount, true);
+  }
+
+  /// Makes the color darker by the given [amount], relatively to the current color luminosity.
+  /// [amount] value must be clamped between 0.0 and 1.0
+  Color darken(double amount) {
+    assert(amount >= 0.0 && amount <= 1.0);
+    return _lightness(amount, false);
+  }
+
+  /// Appends the given [lightness] to the color.
+  /// [lightness] value must be clamped between -1.0 and 1.0
+  Color appendLightness(double lightness) {
+    assert(lightness >= -1.0 && lightness <= 1.0);
+    return _lightness(lightness, null);
+  }
+
+  Color _lightness(double amount, bool? lighten) {
+    final hslColor = HSLColor.fromColor(this);
+
+    switch (lighten) {
+      case true:
+        amount = (1 - hslColor.lightness) * amount;
```

Comment:
why are we multiplying here?

given that lightness is expressed as a relative number from 0.0 to 1.0, i think the "lighten" operation should just add and "darken" subtract the given amount. that is, lighten(0.1) should just +0.1 to the current lightness which equals to a 10% lighter color.

Reply:
Well I multiply because the `amount` represents a percentage of the difference between the actual color lightness and a full white (in `lighten()`) or a full black (in `darken()`). This is the relative method.
In the absolute one, I just take the actual color lightness and I simply add it the given `amount`.

For example taking a color with `0.5` of luminosity, if we lighten it with both methods it results in:

| Amount | Relative | Absolute |
| --- | --- | --- |
| 0.0 | 0.5 | 0.5 |
| 0.1 | 0.55 | 0.6 |
| 0.2 | 0.6 | 0.7 |
| 0.3 | 0.65  | 0.8 |
| 0.4 | 0.7 | 0.9 |
| 0.5 | 0.75 | 1.0 |
| ... | ... | ... |
| 0.7 | 0.85 | 1.0 |
| ... | ... | ... |
| 1.0 | 1.0 | 1.0 |

It is just like in sass, where there is two function to change a color lightness, in a relative way, and an absolute one:
https://sass-lang.com/documentation/modules/color#scale
https://sass-lang.com/documentation/modules/color#lighten

Reply:
the term "scale" seems like a good fit for the multiplying variant. can we use the same terminology? based on the naming, i had expected our "lighten" and "darken" would do the same as in sass but they're now actually scaling instead.

Reply:
Imo this terminology seems more logical 🙂

Reply:
I'd like to finish this work :D

We both agree that the "scale" term fits well for the multiplying/relative variant.
Si I propose to have a single `scaleLightness()`, which accept a range of `-1.0 - 1.0` values.

For the other additive/absolute method, I have mixed opinion.
We could create two `lighten()` and `darken()` functions, but I find this confusing, because the function name isn't enough obvious.
I only have two other naming ideas (which I find better): `appendLightness()` and `adjustLightness()`, which would accept a similar range of value as its relative counterpart.

Wdyt @jpnurmi ?

Reply:
I'm not a color expert but I don't see the problem with `lighten()` and `darken()`. That's how I've been successfully(?) lightening and darkening colors for a very long time. :D

Reply:
maybe just add one `scale({double? hue, double? saturation, double? lightness})` if `lighten()` and `darken()` are not useful for us?

Reply:
> I'm not a color expert but I don't see the problem with `lighten()` and `darken()`. That's how I've been successfully(?) lightening and darkening colors for a very long time. :D

Well, I find the lighten/darken terminology really ambiguous :)
It hides the way it compute the lightness change, which can results in unexpected results.
Btw, in sass, `lighten()` and `darken()` are not included in the new module system.
Instead, they have two new functions: `scale-color()` (relative) and `adjust-color()` (absolute), for color operations.

I really think the sass terminology is the good one.
Imo we should add  `Color.scale()` and `Color.adjust()`.
I can add the hue and saturation support, it can be useful, and it is the same working way as lightness :)

Reply:
ok, sounds like we have a plan! :+1: